### PR TITLE
Adds a keras_recipe on using a keras callback to evaluate a non-tensorflow metric

### PR DESCRIPTION
**Pull request by LukeWood**

### CODE REVIEW

File: examples/keras_recipes/sklearn_metric_callbacks.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+This template can be modified slightly to make it work with any existing sklearn metric.
+"""
+
+import tensorflow as tf
+from sklearn.metrics import jaccard_score
+import numpy as np
+
+
+class JaccardScoreCallback(tf.keras.callbacks.Callback):
```
Comment (fchollet) [inline examples/keras_recipes/sklearn_metric_callbacks.py:35]: Since Keras is imported you can replace all `tf.keras` with just `keras`

File: examples/keras_recipes/sklearn_metric_callbacks.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+print("x_train shape:", x_train.shape)
+print(x_train.shape[0], "train samples")
+print(x_test.shape[0], "test samples")
+
+
+# convert class vectors to binary class matrices
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+
+import tensorflow.keras.layers as layers
```
Comment (fchollet) [inline examples/keras_recipes/sklearn_metric_callbacks.py:95]: Move import at top of file

File: examples/keras_recipes/sklearn_metric_callbacks.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ]
+)
+
+model.summary()
+
+batch_size = 128
+epochs = 15
+
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+summary_writer = tf.summary.create_file_writer("logs/traditional_classifier")
```
Comment (fchollet) [inline examples/keras_recipes/sklearn_metric_callbacks.py:116]: I wonder if this should be moved to the callback constructor, so that all references to the summary API are within the callback class

File: examples/keras_recipes/sklearn_metric_callbacks.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
+"""
+Title: Evaluationg and Exporting scikit-learn Metrics in a Keras callback
+Author: [lukewood](https://lukewood.xyz)
+Date created: 10/07/2021
+Last modified: 10/07/2021
+Description: Example shows how to use Keras callbacks to evaluate and export non-TensorFlow based metrics.
+"""
+"""
```
[the remark this answers is not in the document]
Comment (LukeWood) [reply examples/keras_recipes/sklearn_metric_callbacks.py:8]: Done

File: examples/keras_recipes/sklearn_metric_callbacks.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            )
+            self.summary_writer.flush()
+
+
+"""
+## Sample Usage
+Let's test our `JaccardScoreCallback` class with a real keras model.
+"""
+
+import tensorflow.keras as keras
```
[the remark this answers is not in the document]
Comment (LukeWood) [reply examples/keras_recipes/sklearn_metric_callbacks.py:74]: Done

File: examples/keras_recipes/sklearn_metric_callbacks.py (hunk 2 above)
Comment (LukeWood) [reply examples/keras_recipes/sklearn_metric_callbacks.py:98]: Done

File: examples/keras_recipes/sklearn_metric_callbacks.py (hunk 3 above)
Comment (LukeWood) [reply examples/keras_recipes/sklearn_metric_callbacks.py:119]: Done - yeah this is very nice and matches the tensorboard callback!
